websites/F/Free Web Novel/presence.ts

```diff
@@ -0,0 +1,196 @@
+const presence = new Presence({
+		clientId: "965294297048023050"
+	}),
+	browsingTimestamp = Math.floor(Date.now() / 1000);
+
+presence.on("UpdateData", async () => {
+	let [showCover, showButtons, showBook, showLogo] = await Promise.all([
+		presence.getSetting<boolean>("showCover"),
+		presence.getSetting<boolean>("showButtons"),
+		presence.getSetting<boolean>("showBook"),
+		presence.getSetting<boolean>("logoOnly")
+	]);
+	const [privacy, showTimestamp, showReading] = await Promise.all([
+			presence.getSetting<boolean>("privacy"),
+			presence.getSetting<boolean>("showTimestamp"),
+			presence.getSetting<boolean>("showReading")
+		]),
+		presenceData: PresenceData = {
+			largeImageKey: "nocover"
+		},
+		{ pathname } = window.location,
+		pathnames = [
+			"/",
+			"/history/",
+			"/latest-release-novel/",
+			"/latest-novel/",
+			"/completed-novel/",
+			"/most-popular-novel/",
+			"/search/",
+			"/contact.html",
+			"/sitemap.xml",
+			"/privacy-terms-of-use.html",
+			"/genre/",
+			"/author/"
+		];
+	if (!showReading) {
+		showBook = false;
+		showCover = false;
+		showLogo = true;
+		showButtons = false;
+	} else {
+		presenceData.smallImageKey = "closed";
+		presenceData.smallImageText = "Not Reading";
+	}
+	if (showTimestamp) presenceData.startTimestamp = browsingTimestamp;
+	else delete presenceData.startTimestamp;
+	if (!showBook) showCover = false;
+	switch (true) {
+		case privacy: {
+			presenceData.largeImageKey = "incognito";
+
+			if (
+				(pathnames.includes(pathname) ||
+					document.querySelector<HTMLDivElement>('[class="m-imgtxt"]')) &&
+				showReading
+			)
+				presenceData.details = "Browsing...";
+			else if (
+				document.querySelector<HTMLAnchorElement>(
+					'[title="Read Next chapter"]'
+				) &&
+				showReading
+			) {
+				presenceData.details = "Reading...";
+				presenceData.smallImageKey = "open";
+			}
+			break;
+		}
+		case showLogo: {
+			presenceData.largeImageKey = "fwn_1024";
+			delete presenceData.smallImageKey;
+			delete presenceData.details;
+			delete presenceData.state;
+			delete presenceData.buttons;
+			break;
+		}
+		case pathname === "/": {
+			presenceData.details = "Browsing the homepage";
+			break;
+		}
+		case pathname.includes("/history/"): {
+			presenceData.details = "Browsing recently viewed novels";
+			break;
+		}
+		case pathname.includes("/latest-release-novel/"): {
+			presenceData.details = "Browsing latest released novels";
```

Comment:
Well now I'm confused. Usually when you have the latest of something it's the latest released but I see there's latest novels below so what are these?

Reply:
> Well now I'm confused. Usually when you have the latest of something it's the latest released but I see there's latest novels below so what are these?

Idk man, just going with what the site got to offer and throwing it in here. Ig latest released are novels that got more chapters added to them, cuz looking at the latest novels they got no reviews so I assume they're just new.

Book that's under the latest released page:
![image](https://user-images.githubusercontent.com/96930330/169286121-74d3fcc6-f12f-4708-836c-1fe3ff6c5b56.jpeg)

Reply:
That I understand, but in that case, what are the latest novels?

Reply:
Maybe the latest released are the more popular ones...? Idfk man I'm just taking what's on the site 😂, u need me to do anything with it?